### PR TITLE
fix(homepage-video): add 5 sec preview video for a11y

### LINKED ISSUE #355: Pause, Stop, or Hide Animations (Video backgrounds)
## Detailed description

WCAG has a requirement to [Pause, Stop or Hide](https://www.w3.org/TR/WCAG21/#pause-stop-hide) moving, blinking or scrolling content.  All of the video background content on the site should either stop after 5 seconds or have a control that will allow the user to pause it.  

Two places that I've spotted this so far are the [IBM Design](https://www.ibm.com/design/) page (the "video background" of the video on the landing page) and the [IBM Design Language](https://www.ibm.com/design/language/) page.

This page has a section on [integrating accessibility in video backgrounds](http://thenewcode.com/777/Create-Fullscreen-HTML5-Page-Background-Video) that may be helpful.

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #355

### CODE REVIEW

File: src/content/index.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ people we serve.</span>
 <column lg="16">
 
 <video>
-  <iframe title="IBM Design sizzle video" src="https://player.vimeo.com/video/304672438?muted=1&autoplay=1&loop=1?title=0&byline=0&portrait=0?color=ff0000" width="500" height="281" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen />
+  <iframe id="homepage-video" title="IBM Design sizzle video" src="https://player.vimeo.com/video/329866992?&muted=1&loop=0&autoplay=1" allow="autoplay" width="500" height="281" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen />
```
Comment (asudoh) [inline src/content/index.md]: @jnm2377 Wondering if putting this `<iframe>` into `<Video>` itself is an option - In this way we can easily grab the `<iframe>` without needing to set `id` here. Any thoughts?

File: src/components/Video/Video.js [hunk 2]
```diff
@@ -2,60 +2,81 @@ import React from 'react';
...
   componentDidMount() {
-    const iframe = document.querySelector('iframe');
-    const player = new Player(iframe);
-    player.on('ended', this.onEnd);
-    player.on('pause', this.mitigateKeyboard);
+    const iframe = document.querySelector('#homepage-video');
+    this.player = new Player(iframe);
+    this.player.on('ended', this.onEnd);
+    this.player.setLoop(false);
   }
```
Comment (asudoh) [inline src/components/Video/Video.js]: For better lifecycle support, good to add below:

```javascript
componentWillUnmount() {
  if (this.player) {
    this.player.destroy();
    this.player = null;
  }
}
```

File: src/components/Video/Video.js [hunk 3]
```diff
@@ -2,60 +2,81 @@ import React from 'react';
...
-    const video = document.querySelector('.ibm--video-wrapper');
-    const iframe = document.querySelector('iframe');
-    const player = new Player(iframe);
-    if (!video.classList.contains('active')) {
-      this.onClick();
-    } else {
-      player.pause();
+  // Only plays second video if key = spacebar & it's currently first video
+  onKeyDown = evt => {
+    if (evt.which === 32) {
```
[the remark this answers is not in the document]
Comment (jnm2377) [reply src/components/Video/Video.js]: Ah, good idea! 💯

File: src/content/index.md (hunk 1 above)
Comment (jnm2377) [reply src/content/index.md]: @asudoh like just removing `<iframe>` from the markdown and placing it where we currently render `{children}` within the component, and then select it like `document.querySelector('.ibm--video-wrapper iframe')`?

Comment (asudoh) [reply src/content/index.md]: @jnm2377 Yeah close - If `<Video>` renders the `<iframe>` then we can `<iframe ref={this.iframe}>`. `this.iframe` should be set at around L16 as `this.iframe = React.createRef();`. In this way, we can do `this.player = new Player(this.iframe.current)` at L23.

Comment (jnm2377) [reply src/content/index.md]: Ah I see what you mean. I tried it out and it worked correctly, but the component itself was no longer rendering in a `gatsby-resp-iframe-wrapper` anymore, which altered the video size/responsive styles it was getting from that and making the `iframe` small.

File: src/components/Video/Video.js [hunk 4]
```diff
@@ -141,9 +152,16 @@ export default class Video extends React.Component {
...
+      ' active': this.state.isVideoWrapperActive,
+    });
 
     return (
-      <div className="ibm--video-wrapper" onClick={this.onClick}>
+      <div
+        className={videoWrapperClassName}
+        onClick={this.handlePlaySecondVideo}
+        onKeyDown={this.handlePlaySecondVideo}
+        tabindex="0">
```
Comment (asudoh) [inline src/components/Video/Video.js]:
```suggestion
        tabIndex="0">
```

Also need to replace `class` with `className` in some SVG elements - Just found them looking at the console.

File: src/content/index.md (hunk 1 above)
Comment (asudoh) [reply src/content/index.md]: Got it - Seems that https://www.gatsbyjs.org/packages/gatsby-remark-responsive-iframe/ doesn't work if the `<iframe>` is in a component. In such case, we can grab the DOM ref of the container and search for the `<iframe>` in it, like:

```diff
--- a/src/components/Video/Video.js
+++ b/src/components/Video/Video.js
@@ -14,12 +14,14 @@ export default class Video extends React.Component {
     this.handlePlaySecondVideo = this.handlePlaySecondVideo.bind(this);
   }
 
+  _container = React.createRef();
+
   state = {
     isVideoWrapperActive: false,
   };
 
   componentDidMount() {
-    const iframe = document.querySelector('#homepage-video');
+    const iframe = this._container.current.querySelector('iframe');
     this.player = new Player(iframe);
     this.player.on('ended', this.onEnd);
     this.player.setLoop(false);
@@ -158,6 +160,7 @@ export default class Video extends React.Component {
 
     return (
       <div
+        ref={this._container}
         className={videoWrapperClassName}
         onClick={this.handlePlaySecondVideo}
         onKeyDown={this.handlePlaySecondVideo}
--- a/src/content/index.md
+++ b/src/content/index.md
@@ -13,7 +13,7 @@ people we serve.</span>
 <column lg="16">
 
 <video>
-  <iframe id="homepage-video" title="IBM Design sizzle video" src="https://player.vimeo.com/video/329866992?&muted=1&loop=0&autoplay=1" allow="autoplay" width="500" height="281" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen />
+  <iframe title="IBM Design sizzle video" src="https://player.vimeo.com/video/329866992?&muted=1&loop=0&autoplay=1" allow="autoplay" width="500" height="281" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen />
 </video>
 
 </column>
```

**Pull request merged**